models/trigger_map_item.go

```diff
@@ -155,75 +122,446 @@ func (triggerItem TriggerMapItemModel) MatchWithParams(pushBranch, prSourceBranc
+		}
+		item.CommitMessage = value
 	}
 
-	if triggerItem.PullRequestSourceBranch != "" || triggerItem.PullRequestTargetBranch != "" {
-		if str != "" {
-			str += " "
+	mapInterface, ok = item.ChangedFiles.(map[interface{}]interface{})
+	if ok {
+		value, err := castInterfaceKeysToStringKeys(idx, "changed_files", mapInterface)
+		if err != nil {
+			return TriggerMapItemModel{}, err
 		}
+		item.ChangedFiles = value
+	}
 
-		if triggerItem.PullRequestSourceBranch != "" {
-			str += fmt.Sprintf("pull_request_source_branch: %s", triggerItem.PullRequestSourceBranch)
+	mapInterface, ok = item.Tag.(map[interface{}]interface{})
+	if ok {
+		value, err := castInterfaceKeysToStringKeys(idx, "tag", mapInterface)
+		if err != nil {
+			return TriggerMapItemModel{}, err
 		}
-		if triggerItem.PullRequestTargetBranch != "" {
-			if triggerItem.PullRequestSourceBranch != "" {
-				str += " && "
-			}
+		item.Tag = value
+	}
 
-			str += fmt.Sprintf("pull_request_target_branch: %s", triggerItem.PullRequestTargetBranch)
+	mapInterface, ok = item.PullRequestSourceBranch.(map[interface{}]interface{})
+	if ok {
+		value, err := castInterfaceKeysToStringKeys(idx, "pull_request_source_branch", mapInterface)
+		if err != nil {
+			return TriggerMapItemModel{}, err
 		}
+		item.PullRequestSourceBranch = value
+	}
 
-		str += fmt.Sprintf(" && draft_pull_request_enabled: %v", triggerItem.IsDraftPullRequestEnabled())
+	mapInterface, ok = item.PullRequestTargetBranch.(map[interface{}]interface{})
+	if ok {
+		value, err := castInterfaceKeysToStringKeys(idx, "pull_request_target_branch", mapInterface)
+		if err != nil {
+			return TriggerMapItemModel{}, err
+		}
+		item.PullRequestTargetBranch = value
 	}
 
-	if triggerItem.Tag != "" {
-		if str != "" {
-			str += " "
+	mapInterface, ok = item.PullRequestLabel.(map[interface{}]interface{})
+	if ok {
+		value, err := castInterfaceKeysToStringKeys(idx, "pull_request_label", mapInterface)
+		if err != nil {
+			return TriggerMapItemModel{}, err
+		}
+		item.PullRequestLabel = value
+	}
+
+	return item, nil
+}
+
+func (item TriggerMapItemModel) Validate(idx int, workflows, pipelines []string) ([]string, error) {
+	warnings, err := item.validateTarget(idx, workflows, pipelines)
+	if err != nil {
+		return warnings, err
+	}
+
+	if item.Pattern != "" {
+		if err := item.validateLegacyItemType(idx); err != nil {
+			return warnings, err
 		}
+	} else {
+		if err := item.validateType(idx); err != nil {
+			return warnings, err
+		}
+		if err := item.validateConditionValues(idx); err != nil {
+			return warnings, err
+		}
+	}
+
+	return warnings, nil
+}
 
-		str += fmt.Sprintf("tag: %s", triggerItem.Tag)
+func (item TriggerMapItemModel) validateTarget(idx int, workflows, pipelines []string) ([]string, error) {
+	var warnings []string
+
+	// Validate target
+	if item.PipelineID != "" && item.WorkflowID != "" {
+		return warnings, fmt.Errorf("both pipeline and workflow are defined as trigger target for the %d. trigger item", idx+1)
```

Comment:
```suggestion
		return warnings, fmt.Errorf("trigger #%d: both pipeline and workflow are defined as trigger target", idx+1)
```

Reply:
Great suggestion, updated all the related error messages here: https://github.com/bitrise-io/bitrise/pull/932/commits/a6aed9e86250f831b97df8f0c5a4ef17e287fa72